var/spack/repos/builtin/packages/mofem-cephas/package.py

```diff
@@ -0,0 +1,106 @@
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class MofemCephas(CMakePackage):
+    """mofem-cephas core library"""
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://bitbucket.org/likask/mofem-cephas.git"
+
+    maintainers = ['likask']
+
+    version('0.8.3', git='https://bitbucket.org/likask/mofem-cephas.git',
+        tag='v0.8.3', submodules=True)
+    version('develop',
+        git='https://bitbucket.org/likask/mofem-cephas.git',
+        branch='develop')
+
+    # This option can be only used for development of core lib
+    variant('copy_user_modules', default=True,
+        description='Copy user modules directory instead linking to source')
+    variant('adol-c', default=True, description='Compile with Adol-C')
+    variant('tetgen', default=True, description='Compile with Tetgen')
+    variant('med', default=True, description='Compile with Med')
+    variant('slepc', default=True, description='Compile with Slepc')
+    variant('doxygen', default=False, description='Install doxygen')
+
+    depends_on("mpi")
+    depends_on("parmetis")
+    # Fixed version of hdf5, to remove some problems with dependent
+    # packages, f.e. MED format
+    depends_on("hdf5@:1.8.19+hl+mpi")
+    depends_on("petsc@:3.9.2+mumps+mpi")
+    depends_on('slepc', when='+slepc')
+    depends_on("moab")
+    # Upper bound set to ADOL-C until issues with memory leaks
+    # for versions 2.6: fully resolved
+    depends_on("adol-c@2.5.2~examples", when="+adol-c")
+    depends_on("tetgen", when="+tetgen")
+    depends_on("med", when='+med')
+    depends_on('doxygen+graphviz', when='+doxygen')
+    depends_on('graphviz', when='+doxygen')
+
+    extendable = True
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        return 'mofem'
+
+    def cmake_args(self):
+        spec = self.spec
+        options = []
+
+        # obligatory options
+        options.extend([
+            '-DWITH_SPACK=1',
+            '-DPETSC_DIR=%s' % spec['petsc'].prefix,
+            '-DPETSC_ARCH=',
+            '-DMOAB_DIR=%s' % spec['moab'].prefix])
+
+        # variant packages
+        if '+adol-c' in spec:
+            options.extend(['-DADOL-C_DIR=%s' % spec['adol-c'].prefix])
```

Comment:
here and below no need for `extend` as you add a single element, just do 
```
options.append('-DADOL-C_DIR=%s' % spec['adol-c'].prefix)
```

Reply:
If I change this to extend it is not working,
```
==> Building mofem-cephas [CMakePackage]
==> Executing phase: 'cmake'
==> Error: TypeError: expected string or buffer

/Users/likask/MyBuild/spack/lib/spack/spack/build_systems/cmake.py:231, in cmake:
        228        options = [os.path.abspath(self.root_cmakelists_dir)]
        229        options += self.std_cmake_args
        230        options += self.cmake_args()
  >>    231        with working_dir(self.build_directory, create=True):
        232            inspect.getmodule(self).cmake(*options)
```

Reply:
> If I change this to extend it is not working,

`append`, `not extend`!

Reply:
```
list.append(obj)
```

Reply:
Sorry my mistake, I change this to **append** and get above error.

Reply:
> Sorry my mistake, I change this to append and get above error.

can you please push the changes?

Reply:
If you do append, you get something like that,
```
 -DWITH_SPACK=1
    -DPETSC_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/petsc-3.9.2-qdi36dezfsbza3a3zpsz4ip2maivk2b2
    -DPETSC_ARCH=
    -DMOAB_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/moab-5.0.0-mrkqrjsd6ih5e2pwqhccbrjrcvaimsae
    ['-DADOL-C_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/adol-c-2.5.2-ahoqqihx3zo72rmlymhuj3w5yuvkv34q']
    -DTETGEN_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/tetgen-1.5.0-bfxmmbgp2d4rychv4nsgl2a6gsfqgeyb
    -DMED_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/med-3.2.0-6qucn4cb337mh64ptxp7hjdaea25p6sx
    -DSLEPC_DIR=/Users/likask/MyBuild/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/slepc-3.9.1-ucwwfpjupgb7yd72cuejbkvbhsuuy7bz
    -DSTAND_ALLONE_USERS_MODULES=YES
```

Reply:
Sorry, I see my error. My python is very basic.

Reply:
@likask you probably forgot to remove brackets, please have another look at the post above.

Reply:
> Sorry, I see my error. My python is very basic.

no worries.